motionplan/planManager.go

```diff
@@ -305,18 +316,29 @@ func (pm *planManager) planParallelRRTMotion(
 ) {
 	var err error
 	// If we don't pass in pre-made maps, initialize and seed with IK solutions here
-	if maps == nil {
-		planSeed := initRRTSolutions(ctx, pathPlanner, seed)
-		if planSeed.planerr != nil || planSeed.steps != nil {
-			solutionChan <- planSeed
-			return
+	if !pm.useTPspace {
+		if maps == nil {
+			planSeed := initRRTSolutions(ctx, pathPlanner, seed)
+			if planSeed.planerr != nil || planSeed.steps != nil {
+				solutionChan <- planSeed
+				return
+			}
+			maps = planSeed.maps
+		}
+	} else {
+		if maps == nil {
+			startNode := &basicNode{q: make([]referenceframe.Input, len(pm.frame.DoF())), pose: spatialmath.NewZeroPose()}
+			goalNode := &basicNode{q: make([]referenceframe.Input, len(pm.frame.DoF())), pose: goal}
+			maps = &rrtMaps{
+				startMap: map[node]node{startNode: nil},
+				goalMap:  map[node]node{goalNode: nil},
+			}
 		}
-		maps = planSeed.maps
 	}
 
 	// publish endpoint of plan if it is known
 	var nextSeed node
-	if len(maps.goalMap) == 1 {
+	if maps != nil && len(maps.goalMap) == 1 {
```

Comment:
At this point maps should never be nil by my estimation

Reply:
Good point